src/signals/incident-management/containers/IncidentDetail/components/Detail/components/Location/index.js

```diff
@@ -1,67 +1,78 @@
+const Description = styled.dd`
+  position: relative;
+  display: flex;
+`;
 
-const Location = ({
-  incident, stadsdeelList, onShowLocation, onEditLocation,
-}) => {
-  const address = incident.location.address;
-  return (
-    <dl className="location">
-      <dt className="location__definition" data-testid="location-definition">Locatie</dt>
-      <dd className="location__value">
-        <button
-          className="location__value-button-edit incident-detail__button--edit"
-          type="button"
-          onClick={onEditLocation}
-          data-testid="location-button-edit"
-        ></button>
-        <button
-          className="location__value-button-show"
-          type="button"
-          onClick={onShowLocation}
-          data-testid="location-button-show"
-        >
-          <MapDetail
-            value={incident.location}
-            hideAttribution
-            hideZoomControls
-            useSmallMarker
-            zoom="15"
-          />
-        </button>
-        {incident.location.address_text
-          ? (
-            <div className="location__value-address">
-              <div data-testid="location-value-address-stadsdeel">
-                Stadsdeel: {getListValueByKey(stadsdeelList, incident.location.stadsdeel)}
-              </div>
-              <div data-testid="location-value-address-street">
-                {address.openbare_ruimte} {address.huisnummer}
-                {address.huisletter}
-                {address.huisnummer_toevoeging ? `-${address.huisnummer_toevoeging}` : ''}
-              </div>
-              <div data-testid="location-value-address-city">
-                {address.postcode} {address.woonplaats}
-              </div>
-            </div>
-          )
-          : <span data-testid="location-value-pinned">Locatie is gepind op de kaart</span>}
-      </dd>
-    </dl>
-  );
-};
+const EditButton = styled(Button)`
+  position: absolute;
+  right: 0;
+  top: 0;
+`;
+
+const Address = styled.div`
+  margin-left: ${themeSpacing(12)};
+`;
+
+const Location = ({ incident: { location }, onShowLocation, onEditLocation }) => (
+  <Fragment>
+    <dt data-testid="detail-location">Locatie</dt>
+    <Description>
+      <EditButton
+        variant="application"
+        iconSize={20}
+        icon={<IconEdit />}
+        type="button"
+        onClick={onEditLocation}
+        data-testid="location-button-edit"
+      />
+
+      <MapTrigger onClick={onShowLocation} data-testid="location-button-show">
+        <MapDetail value={location} hideAttribution hideZoomControls useSmallMarker zoom="15" />
+      </MapTrigger>
+
+      {location.address_text ? (
+        <Address>
+          <div data-testid="location-value-address-stadsdeel">
+            Stadsdeel: {getListValueByKey(stadsdeelList, location.stadsdeel)}
+          </div>
+          <div data-testid="location-value-address-street">
+            {location.address.openbare_ruimte} {location.address.huisnummer}
```

Comment:
Are we sure that the location properties exist?

Reply:
Good one; I've made sure that address parts are conditionally rendered